experimental/PlaneCurve/EllCurveZnZ.jl

```diff
@@ -0,0 +1,704 @@
+export sum_Point_EllCurveZnZ,
+    ECM,
+    rand_pair_EllCurve_Point,
+    IntMult_Point_EllCurveZnZ,
+    cornacchia_algorithm,
+    Miller_Rabin_test,
+    Pollard_rho,
+    Pollard_p_1,
+    ECPP
+
+################################################################################
+# Elliptic curves over a ring Z/nZ
+################################################################################
+################################################################################
+# Helping functions
+################################################################################
+# addition on two projective points on an elliptic curve over Z/nZ, given in
+# short Weierstrass form, when it is possible. It gives as an output a pair
+# composed of the coordinates of the sum and ZZ(1) if the sum exists, and
+# (0, 0, 0) and the gcd otherwise.
+# Adapted from the corresponding code in Hecke.
+
+function _add(
+    P::Array{Nemo.fmpz_mod,1},
+    Q::Array{Nemo.fmpz_mod,1},
+    E::Array{Nemo.fmpz_mod,1},
+)
+    length(P) == 3 && length(Q) == 3 || error("arrays of size 3 required")
+    length(E) == 2 || error("array of size 2 required")
+    A = P[1].parent
+    A == Q[1].parent && A == E[1].parent || error("Not the same parent")
```

Comment:
Is this enough? Don't you also have to check `P[2]`, `P[3]`, `Q[2]`, `Q[3]`, `E[2]`? Perhaps like this:
```suggestion
    all(x -> parent(x) == A, [P...,Q...,E...]) || error("Not the same parent")
```